### PR TITLE
Beta Tester: Do not run code in tracks debug unless WC_ABSPATH is defined.

### PR DESCRIPTION
**Pull request by samueljseay**
### Changes proposed in this Pull Request:

Fix 2 bugs that cause beta tester to crash:

1. Fix an edge case in running code of the beta tester that depends on WC_ABSPATH being defined.

Between activations of versions of WC that beta tester installs, the PHP code of the beta tester can run and in most cases I removed dependencies on WC itself to avoid that happening or made sure the code does not run if WC is not installed. But in this case I checked for WooCommerce class existing, but technically that's not robust enough.

If the class exists but hasn't been instantiated (which is when it creates WC_ABSPATH) and this code runs first then it will error as it seems to have in https://github.com/woocommerce/woocommerce/issues/39441

2. Fix a bug where if beta tester is installed without WooCommerce it crashes. This was due to loading a class that is not loaded when WooCommerce is not present. We now when registering scripts for live branches avoid doing that work altogether if Woo is not installed and we rely on the error message to inform users to install WooCommerce. In future we can remove all dependence on WooCommerce from beta tester but for now this doesn't crash the plugin. This solves https://github.com/woocommerce/woocommerce/issues/38804

### How to test the changes in this Pull Request:

1. For the first bug you can create an AT site to ensure it doesn't crash. Build a zip of the beta tester plugin from this branch (from the project subdir run `pnpm build:zip`

2. Install the plugin zip on your AT site alongside WooCommerce. Activate it. It should not crash.

3. For the second bug, remove beta tester, then remove WooCommerce just to ensure your AT site is in clean state.

4. Now just install and activate beta tester without WooCommerce.

5. It should not crash and you should see a error message stating you need WooCommerce installed to use beta tester.

### Changelog entry

-   [x] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [x] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 
Fix a bug where WC_ABSPATH is used when the WooCommerce instance might not have defined it yet.

#### Comment 

</details>

### CODE REVIEW

**Comment by samueljseay** [comment]
Hey @psealock I noticed @lsinger is not around. Can you take a look at this one?

File: plugins/woocommerce-beta-tester/includes/class-wc-beta-tester-live-branches.php [hunk 1]
```diff
@@ -15,10 +15,23 @@ class WC_Beta_Tester_Live_Branches {
...
 		add_action( 'admin_init', array( $this, 'register_scripts' ) );
 	}
 
+	/**
+	 * Check if WooCommerce is installed.
+	 *
+	 * @return bool - True if WooCommerce is installed, false otherwise.
+	 */
+	private function woocommerce_is_installed() {
+		return class_exists( 'WooCommerce' );
```
Comment (psealock) [inline plugins/woocommerce-beta-tester/includes/class-wc-beta-tester-live-branches.php:32]: If this check wasn't sufficient in the case of `plugins/woocommerce-beta-tester/api/tracks/class-tracks-debug-log.php`, why not just directly make this a check for `defined( 'WC_ABSPATH' )`?

**Review by psealock — COMMENTED** [review]
This is testing well. Just a question of logic below

File: plugins/woocommerce-beta-tester/includes/class-wc-beta-tester-live-branches.php (hunk 1 above)
Comment (samueljseay) [reply plugins/woocommerce-beta-tester/includes/class-wc-beta-tester-live-branches.php:32]: @psealock the log code has that flaw because it asks if woocommerce class exists then tries to access a constant defined long after the class exists. You can see how the order of checks is wrong. That class needs refactoring or the constant loaded earlier. We can refactor at some point. I would rather not have to check the constant exists because those constants are imo more of a private api that could change.

This code doesnt have that flaw. It doesnt use that late defined constant and checking if woocommerce class exists is in 99% of cases the more sensible check in my opinion. This code will never break in that way because if woo doesnt exist we never do anything. 

Tldr; one is a patch to an issue i dont want to refactor immediately. the other is doing things the more proper way imo.

**Review by psealock — APPROVED** [review]
Thanks for the background info
